### ISSUE TITLE
`Wweak-vtables` should not trigger in a module

### ISSUE THREAD

**Issue by davidstone** [llvm/llvm-project#193004]
Given the following translation unit

```c++
export module m;

struct s {
	virtual void f() {
	}
};
```

clang warns

```console
<source>:3:8: warning: 's' has no out-of-line virtual method definitions; its vtable will be emitted in every translation unit [-Wweak-vtables]
    3 | struct s {
      |        ^
1 warning generated.
Compiler returned: 0
```

I expect no warnings in this case.

See it live: https://godbolt.org/z/djvjMv8z1

Related: https://github.com/llvm/llvm-project/issues/62465

**Comment by davidmenggx** [comment]
I can give this issue a try
